### ISSUE TITLE
Queue needs to manage scratch!

### ISSUE THREAD

**Issue by raimis** [Acellera/htmd#491]
By default, *Psi4* uses `/tmp` for anything that does not fit into RAM. In case of our own infrastructure, it has be changed to a different path. Ideally, a queue object should handle that and pass to *Psi4*.

We need something like that working:
```
>>> queue.scratch
'/huge_disk/tmp'
```

**Comment by j3mdamas** [comment]
Slurm: https://help.rc.ufl.edu/doc/Temporary_Directories
LSF: https://scicomp.ethz.ch/wiki/Using_the_batch_system#Scratch_space

We need to implement these to work with Psi4/Gaussian.

**Comment by raimis** [comment]
With *Psi4*, you just need to set `PSI_SCRATCH` (http://www.psicode.org/psi4manual/master/external.html#envvar-PSI_SCRATCH)

**Comment by tonigi** [comment]
I think the setenv can be put in the .sh file, since there are other env settings already in there. Please don't put it in the input file because then it becomes system-dependent and it can't run on gpugrid.

**Comment by j3mdamas** [comment]
gpugrid does not use slurm or lsf, which are the queues that concern here, so I think it's safe.

**Comment by tonigi** [comment]
i am making psi4 run on gpugrid

**Comment by j3mdamas** [comment]
yeah, I know. I think this needs will be handled at the queue level (htmd/queues), so I think it's OK. What are your concerns?

**Comment by tonigi** [comment]
i was asking not to set the scratch dir in the input file via python. hardcoded paths would fail in gpugrid.

**Comment by j3mdamas** [comment]
I see. When we do it, we make sure the setting is totally associated with the queue and not with the application.

**Comment by raimis** [comment]
The current hack to work on our infrastructure:
```diff
diff --git a/htmd/qm/psi4.py b/htmd/qm/psi4.py
index 0e7ac6f..9b312b5 100644
--- a/htmd/qm/psi4.py
+++ b/htmd/qm/psi4.py
@@ -115,7 +115,7 @@ class Psi4(QMBase):
 
     @property
     def _command(self):
-        return 'export HTMD_PSI4_WORKDIR=$(pwd)\npsi4 -i psi4.in -o psi4.out &> psi4.log'
+        return 'export HTMD_PSI4_WORKDIR=$(pwd)\nexport PSI_SCRATCH=/home/tmp\npsi4 -i psi4.in -o psi4.out &> psi4.log'
 
     def _completed(self, directory):
         return os.path.exists(os.path.join(directory, 'psi4.out'))
```

**Comment by raimis** [comment]
A bit more general hack: 7ab51bf

**Comment by j3mdamas** [comment]
@raimis actually, the general hack only works for Slurm, as LSF does not pass environment variables to the running nodes (that I know about). Still need to address this.

**Comment by mj-harvey** [comment]
In the general case, expect resource management systems to create per-job temporary directories, with paths that are only resolvable at execution time. cf PBS Pro's TMPDIR on cx1

**Comment by j3mdamas** [comment]
thanks @mj-harvey! Yes, in general the queue needs to be solving the temporary directories for us and presenting our applications with their location, and not us deciding for the queue.

In this case, I'm ignoring PBS for now, as I don't have access to such a queueing system, but focusing on Slurm and LSF

**Comment by j3mdamas** [comment]
This is getting closed and replaced by #610
